src/lambda/config/templates.ts

```diff
@@ -33,57 +52,44 @@ export class DefaultLoadTemplatesConfigContext implements LoadTemplatesConfigCon
     public readonly fileExists = fsUtils.fileExists
     public readonly readFile = fsUtils.readFileAsString
 
-    public async saveDocumentIfDirty(editorPath: string): Promise<void> {
-        const path = _path.normalize(vscode.Uri.file(editorPath).fsPath)
-        const document = vscode.workspace.textDocuments.find(doc => {
-            if (!doc.isDirty) {
-                return false
-            }
-
-            if (_path.normalize(doc.uri.fsPath) !== path) {
-                return false
-            }
-
-            return true
-        })
+    public saveDocumentIfDirty = saveDocumentIfDirty
+}
 
-        if (document) {
-            await document.save()
-        }
-    }
+export function getTemplatesConfigPath(workspaceFolderPath: string): string {
+    return _path.join(
+        workspaceFolderPath,
+        '.aws',
+        'templates.json'
+    )
 }
 
 export async function load(
     workspaceFolderPath: string,
     context: LoadTemplatesConfigContext = new DefaultLoadTemplatesConfigContext()
 ): Promise<TemplatesConfig> {
-    const templatesConfigPath = _path.join(
+    const templatesConfigPath = getTemplatesConfigPath(
         workspaceFolderPath,
-        '.aws',
-        'templates.json'
     )
 
+    return await loadTemplatesConfig(templatesConfigPath, context)
+}
+
+async function loadTemplatesConfig(
+    path: string,
+    context: LoadTemplatesConfigContext = new DefaultLoadTemplatesConfigContext()
+): Promise<TemplatesConfig> {
     try {
-        await context.saveDocumentIfDirty(templatesConfigPath)
+        await context.saveDocumentIfDirty(path)
 
-        if (!(await context.fileExists(templatesConfigPath))) {
+        if (!(await context.fileExists(path))) {
             return {
                 templates: {}
             }
         }
 
-        const raw = await context.readFile(templatesConfigPath)
-        const errors: ParseError[] = []
-        const config = parse(raw, errors) as TemplatesConfig
-        if (errors.length > 0) {
-            const message = errors.length === 1 ?
-                ` ${formatParseError(errors[0])}`  :
-                `${os.EOL}${errors.map(formatParseError).join(os.EOL)}`
+        const raw = await context.readFile(path)
 
-            throw new Error(`Could not parse .aws/templates.json:${message}`)
-        }
-
-        return config
+        return loadTemplatesConfigFromJson(raw)
     } catch (err) {
         if (Array.isArray(err) && (err as any[]).length === 1) {
```

Comment:
Use logger, it should be able to handle any number of params

Reply:
Let's look at that in a follow up change